File: java/Strategy/complete/Context.java
public class Context {

  public Strategy strategy;

  //set strategy to passed in value
  public void set_strategy(Strategy strategy){
    this.strategy = strategy;
  }

  //execute the current strategy
  public int execute_strategy(int num1, int num2) throws Exception{
    if(strategy != null)
      return this.strategy.execute(num1, num2);
    else
      throw new Exception("strategy of context must be set before call to execute_strategy");
  }
}

File: python/Strategy/complete/Context.py
from Strategy import Strategy

class Context(object):

    def __init__(self):
        #set the initial strategy
        self.strategy = None

    def change_strategy(self, strat):
        #change current strategy to passed in strat
        self.strategy = strat

    def execute_strategy(self, num1, num2):
        #execute the current stategy on passed in numbers
        return self.strategy.execute(num1, num2)
